src/thumbnailer.cc

```diff
@@ -94,29 +99,34 @@ Thumbnailer::Status Thumbnailer::GenerateAnimation(WebPData* const webp_data) {
   WebPDataInit(&new_webp_data);
 
   while (min_quality <= max_quality) {
-    int middle = (min_quality + max_quality) / 2;
-    config_.quality = middle;
+    int mid_quality = (min_quality + max_quality) / 2;
+    for (auto& frame : frames_) {
+      frame.config.quality = mid_quality;
+    }
 
     if (GenerateAnimationNoBudget(&new_webp_data) != kOk) {
       return kMemoryError;
     }
 
     if (new_webp_data.size <= byte_budget_) {
-      final_quality = middle;
+      final_quality = mid_quality;
       WebPDataClear(webp_data);
       *webp_data = new_webp_data;
-      min_quality = middle + 1;
+      min_quality = mid_quality + 1;
     } else {
-      max_quality = middle - 1;
+      max_quality = mid_quality - 1;
       WebPDataClear(&new_webp_data);
     }
   }
 
-  config_.quality = final_quality;
+  for (auto& frame : frames_) {
+    frame.config.quality = final_quality;
+  }
+
   return (final_quality == -1) ? kByteBudgetError : kOk;
 }
 
-int Thumbnailer::GetPSNR(WebPPicture* const pic, int quality) {
+int Thumbnailer::GetPSNR(WebPPicture* const pic, WebPConfig config) {
```

Comment:
`WebPConfig` is not that big but it might be better to use a reference argument here: `const WebPConfig& config`.
`WebPConfig config` would create a copy every time the `GetPSNR()` function is called (unless the compiler optimizes it out). Using a const reference avoid that but checks that it is not modified within the function (safer).

Summary for arguments:
- Use a pass-by-value argument when the type is trivial (`int some_int`, `Thumbnailer::Status some_enum`).
- Use a constant pass-by-reference argument when the type is not (`const WebPConfig& some_conf`).
- Use a constant pointer to a modifiable value when it is an output (`int* const number_of_stuff`, `WebPConfig* const config_to_set`).